integration-test/src/main/java/com/sequenceiq/it/cloudbreak/util/wait/WaitUtil.java

```diff
@@ -379,136 +414,130 @@ private boolean checkFailedStatuses(EnvironmentStatus currentStatus) {
         }
         if (waitResult == WaitResult.FAILED) {
-            StringBuilder builder = new StringBuilder("The stack has failed: ").append(System.lineSeparator());
-            SdxClusterResponse sdxResponse = getSDXClusterResponse(sdxClient, name);
-            if (sdxResponse != null && sdxResponse.getStatus() != null) {
-                builder.append("statusReason: ").append(sdxResponse.getStatusReason());
-            }
+            StringBuilder builder = new StringBuilder("SDX has failed: ").append(System.lineSeparator());
+            sdxResponse = getSDXClusterResponse(sdxClient, name);
+            status = sdxResponse.getStatus().name();
+            reason = sdxResponse.getStatusReason() != null ? sdxResponse.getStatusReason()
+                    : "SDX Status Reason is not available";
+            builder.append("statusReason: ").append(reason);
             LOGGER.error(builder.toString());
-            throw new TestFailException(builder.toString());
+            errors = Map.of("status", status, "reason", reason);
         } else if (waitResult == WaitResult.TIMEOUT) {
-            SdxClusterResponse sdxResponse = getSDXClusterResponse(sdxClient, name);
-            LOGGER.error("Timeout happened while waiting for {} status at {}, the current status is {}", desiredStatus.name(), name,
-                    sdxResponse.getStatus().name());
-            throw new TestFailException("Timeout happened while waiting for " + desiredStatus.name() + " status at " + name + ", the current status is "
-                    + sdxResponse.getStatus().name());
+            sdxResponse = getSDXClusterResponse(sdxClient, name);
+            status = sdxResponse.getStatus().name();
+            reason = sdxResponse.getStatusReason() != null ? sdxResponse.getStatusReason()
+                    : "SDX Status Reason is not available";
+            LOGGER.error("Timeout happened while waiting for {} status at {}, the current status is {} with reason: {}", desiredStatus.name(),
+                    name, status, reason);
+            errors = Map.of("status", status, "reason", reason);
         } else if (DELETED != desiredStatus) {
-            SdxClusterResponse sdxResponse = getSDXClusterResponse(sdxClient, name);
+            sdxResponse = getSDXClusterResponse(sdxClient, name);
             if (sdxResponse != null) {
-                errors = Map.of("status", sdxResponse.getStatus().name());
+                status = sdxResponse.getStatus().name();
+                reason = sdxResponse.getStatusReason() != null ? sdxResponse.getStatusReason()
+                        : "SDX Status Reason is not available";
+                errors = Map.of("status", status, "reason", reason);
             }
         }
         return errors;
     }
 
-    public SdxTestDto waitForSdxInstanceStatus(SdxTestDto sdxTestDto, SdxClient sdxClient, String hostGroup, InstanceStatus desiredState) {
```

Comment:
By eliminating these methods the original ones got even more complex.

Reply:
This change was required.